### PR TITLE
fix(autocomplete): attach overlay to a more accurate input element

### PR DESCRIPTION
**Pull request by willshowell**
Fixes https://github.com/angular/material2/issues/4914
Fixes https://github.com/angular/material2/issues/5709

I removed the `.mat-autocomplete-panel-below` and `.mat-autocomplete-panel-above` classes since they were only being used for css offsets, but maybe they should stay if users have been querying them?

Additionally, it looks like `positionY: AutocompletePositionY` and `_subscribeToPositionChanges` are no longer needed internally, but thought I should check first before removing parts of the public API .

### CODE REVIEW

File: src/lib/input/input-container.ts [hunk 1]
```diff
@@ -457,6 +457,7 @@ export class MdInputContainer implements AfterViewInit, AfterContentInit, AfterC
 
   /** Reference to the input's underline element. */
   @ViewChild('underline') underlineRef: ElementRef;
+  @ViewChild('connectionContainer') _connectionContainerRef: ElementRef;
```
Comment (mmalerba) [inline src/lib/input/input-container.ts]: Can you not use the `underlineRef`? The datepicker is using it for a similar purpose.

Comment (willshowell) [inline src/lib/input/input-container.ts]: https://github.com/angular/material2/issues/4914#issuecomment-305617121 You still have to add an offset for when the fallback is above

**Comment by mmalerba** [comment]
Also for the no-longer-used bits of the API, lets get rid of them and if we encounter too many problems trying to merge it into Google's codebase we can add them back.

**Comment by willshowell** [comment]
@crisbeto weird... it looks like the 6px/24px offsets are still there. Is that Edge? Chrome 59 macOS shows this:

screen shot 2017-08-07 at 9 55 34 am

screen shot 2017-08-07 at 9 55 43 am

**Comment by crisbeto** [comment]
I looked at it in Chrome 59 or 60 on Windows.

**Comment by willshowell** [comment]
Just checked on Windows and it looked the same to me. Also deployed to https://willshowell-demo.firebaseapp.com/autocomplete

Also note that half of the 2px focused underline is obstructed when panel is below because the underline and ripple are absolutely positioned. I can add back a 1px offset or leave as is and later look into possible underline changes that would fix it.

**Comment by willshowell** [comment]
Could you remove the `merge-ready` label? I'd like to remove a couple more unused lines supporting the old approach, per @mmalerba's suggestion.

**Comment by willshowell** [comment]
Thanks @crisbeto, change made

**Pull request merged**